---
title: "Final Class Project"
author: 'Tanvi Manjrekar, Joy Tumusiime'
date: "2024-05-16"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float: true
    number_sections: true
    theme: spacelab
  pdf_document:
    toc: true
    toc_depth: '3'
---

# Abstract:

Various phylum were found Guanica State Forest including Actinobacteriota which is the most prevalant at this site. Elevation, soil temperatures, and pH are analyzed at this site. Dormibacterota are found in many ecosystem subtypes, but are most prevelant in shrublands and wetlands.The gene count was also analyzed for Dormibacterota. 


# Motivating factors:

The motivating factors for this experiment was to analyze the diversity of taxonomic groups at the Guanica Forest site. Knowing the diversity of taxonomic groups can allow researchers to understand what species can grow in certain areas around the world depending on the geography of the area. Additionally, analyzing the sites in which Dormibacterota was found allows researchers to understand where this certain type of species prefers to live in or thrives in. 

# Introduction:

Guánica State Forest is a subtropical dry forest in southwest Puerto Rico. It is the best preserved dry forest in the Caribbean. It has a warm climate with two rainy/ hurricane seasons. It is home to over 700 species of plants that are divided into three groups: deciduous forest, semi-evergreen forest, and scrub forest. Its most famous plant is a guaiac wood tree that could be as old as 1,000 years. This site is home to multiple different ecosystems including beaches, coral reefs, salt flats, mangrove forests, and limestone caverns. Half of Puerto Rico's birds occur in the Guánica State Forest and it is one of the few habitats where the Cook's pallid anole (lizard species) can be found. This forest has both marine and terrestrial wildlife, including coral reefs, birds, grasshoppers, ants, etc (Guanica).
Dormibacterota is uncultured bacteria that is normally found in cold deserts (Ferrari) and are a phylum of oligotrophic bacteria that live under the soil. They are known for their survival mechanisms that allow them to survive under starvation conditions. They are thought to be aerobic heterotrophs and based on genome analysis, they have been found to synthesize, store, and break down glycogen. This phylum of bacteria is not very well researched since they are most commonly found in extremely cold environments. There is a lot of ongoing research that is looking into Dormibacterota phylogenetic relationships and their contribution to the environment in which they live. 

# Methods:
NEON MAGS Data for our site “Guanica State Forest and Biosphere Reserve '' and our phylum “Dormibacterota '' was analyzed using RStudio and RMarkdown. We imported this NEON MAGs Data into an R Markdown file and used various R markdown codes to organize the data into tables and graphs in order to be analyzed efficiently. 


## Loading Libraries 

```{r}
library(tidyverse)
library(knitr)
library(ggtree)
library(ggimage)
library(rphylopic)
library(treeio)
library(tidytree)
library(ape)
library(TreeTools)
library(phytools)
library(DT)
```

```{r}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON.csv")

head(NEON_MAGs)

str(NEON_MAGs)

NEON_MAGs_Ind <- NEON_MAGs %>% 
  filter(`Genome Name` != "NEON combined assembly") 

NEON_MAGs_Ind_tax <- NEON_MAGs_Ind %>% 
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), "; ", remove = FALSE)
```


```{r}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON.csv") %>% 
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`)) %>% 
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), "; ", remove = FALSE) %>% 
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
   separate(`Genome Name`, c("Site","Sample Name"), " - ") %>%
   mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```

```{r}
NEON_metagenomes <- read_tsv("data/NEON/exported_img_data_Gs0161344_NEON.tsv") %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T))
```

```{r}
NEON_metagenomes <- NEON_metagenomes %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")
```

```{r}
NEON_chemistry <- read_tsv("data/NEON/neon_plot_soilChem1_metadata.tsv") %>% 
  # remove -COMP from genomicsSampleID
  mutate_at("genomicsSampleID", str_replace, "-COMP", "") 
```


```{r}
str(NEON_MAGs)
```


```{r}
NEON_MAGs_metagenomes_chemistry <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>% 
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID"))

NEON_data_Dormibacterota <- NEON_MAGs %>% 
 filter(str_detect(Phylum, "Dormibacterota"))

```

```{r}
NEON_data_GUAN <- NEON_MAGs %>% 
  filter(str_detect(Site, "Guanica State Forest and Biosphere Reserve"))
```

```{r}
NEON_chemistry_GUAN <- NEON_chemistry %>% 
 filter(str_detect(siteID, "GUAN"))
```


```{r}
tree_arc <- read.tree("data/NEON/gtdbtk.ar53.decorated.tree")
tree_bac <- read.tree("data/NEON/gtdbtk.bac120.decorated.tree")
```

```{r}
node_vector_bac = c(tree_bac$tip.label,tree_bac$node.label)

grep("Dormibacterota", node_vector_bac, value = TRUE)
```

```{r}
match(grep("Dormibacterota", node_vector_bac, value = TRUE), node_vector_bac)
```

```{r}
tree_bac_node_Dormibacterota <- Preorder(tree_bac)
tree_Dormibacterota <- Subtree(tree_bac_node_Dormibacterota, 1712)
```

```{r}
NEON_MAGs_ind <- NEON_MAGs %>% 
  filter(`Assembly Type` == "Individual") 
```

```{r}
NEON_MAGs_co <- NEON_MAGs %>% 
  filter(`Assembly Type` == "Combined")
```

```{r}
# Select the GTDB Taxonomic lineage and separate into taxonomic levels
sankey_data <- NEON_MAGs_co %>% 
  select(`GTDB-Tk Taxonomy Lineage`) %>% 
  # NAs are likely Archaea
  replace_na(list(`GTDB-Tk Taxonomy Lineage` = 'Archaea')) %>% 
  # Pavian format requires p__ etc
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), "; ") 
```

```{r}
NEON_MAGs_co_2 <- NEON_MAGs_co %>%
  filter(`Phylum` == "Dormibacterota")
```

```{r}
# Select the GTDB Taxonomic lineage and separate into taxonomic levels
sankey_data <- NEON_MAGs_co_2 %>% 
  select(`GTDB-Tk Taxonomy Lineage`) %>% 
  # NAs are likely Archaea
  replace_na(list(`GTDB-Tk Taxonomy Lineage` = 'Archaea')) %>% 
  # Pavian format requires p__ etc
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), "; ")
```

```{r}
NEON_MAGs_2 <- NEON_MAGs %>%
  filter(`Site ID` == "GUAN")
```

```{r}
# Select the GTDB Taxonomic lineage and separate into taxonomic levels
sankey_data <- NEON_MAGs_2 %>% 
  select(`GTDB-Tk Taxonomy Lineage`) %>% 
  # NAs are likely Archaea
  replace_na(list(`GTDB-Tk Taxonomy Lineage` = 'Archaea')) %>% 
  # Pavian format requires p__ etc
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), "; ") 
```

```{r}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON_edArchaea.csv") %>% 
  # remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`)) %>% 
  # create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), "; ", remove = FALSE) %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```


## Results

### Bacterial genomes at Guanica State Forest and Biosphere Reserve

```{r}
NEON_data_GUAN %>% 
ggplot(aes(x = Phylum)) +
  geom_bar(position = position_dodge2(width = 0.9, preserve = "single")) +
  coord_flip() +
  facet_wrap(vars(Site), scales = "free", ncol = 2)
  labs(title = "Taxonomic Groups Found at Guanica State Forest", y = "Taxonomix Groups", x= "Count")
```

This graph showcases the different amounts of phylum found at Guanica State Forest.

```{r}
NEON_data_GUAN %>% 
  filter(is.na(Phylum) | is.na(Class) | is.na(Order) | is.na(Family) | is.na(Genus)) %>% 
ggplot(aes(x= `Site ID`)) +
  geom_bar(position = position_dodge2(width = 0.9, preserve = "single")) +
  coord_flip() +
  labs(title = "Novel Bacteria Discovered at Guanica State Forest", x = "Site ID", y = "Count")
```

This graph showcases the amount of novel bacteria discovered at Guanica State Forest.

```{r}
NEON_data_GUAN %>% 
ggplot(aes(x = fct_rev(fct_infreq(`Site ID`)), fill = Phylum)) +
  geom_bar() +
  coord_flip() +
  labs(title = "Total Number of MAGS at Guanica State Forest", x = "Site ID", y = "Count")
```

This graph displays the total number of MAGS of each phylum present at Guanica State Forest 

```{r}
NEON_chemistry_GUAN %>% 
  ggplot(aes(x=elevation, y=soilTemp))+
  geom_point()+
  labs(title = "Relationship between soil temperature and elevation at Guanica State Forest", y = "Soil Temperature", x= "Elevation (m)")
```

This graph showcases the relationship between soil temperature and elevation at Guanica State Forest. There seems to be relationship between these.


```{r}
NEON_chemistry_GUAN %>% 
  ggplot(aes(x=elevation, y=soilInCaClpH))+
  geom_point()+
  labs(title = "Relationship between the pH of soil in CaCL and elevation at Guanica State Forest", y = "soilInCaClpH", x= "Elevation (m)")
```

This graph showcases the relationship between the pH of soil in CaCL and elevation at Guanica State Forest. There seems to be no relationship between these.

```{r}
NEON_chemistry_GUAN %>% 
  ggplot(aes(x=soilTemp, y=soilInWaterpH))+
  geom_point()+
  labs(title = "Relationship between the soil temperature and pH of Soil in Water at Guanica State Forest", y = "soilInWaterpH", x= "soilTemp")
```

This graph showcases the relationship between the soil temperature and pH of soil in water at Guanica State Forest. There seems to be a positive correlation linear relationship between these with one outlier.

```{r}
NEON_chemistry_GUAN %>% 
  ggplot(aes(x=decimalLatitude, y=soilTemp))+
  geom_point()+
  labs(title = "Relationship between the Decimal Latitude and Soil Temperature at Guanica State Forest", y = "Soil Temperature", x= "Decimal Latitude")
```
This graph showcases the relationship between the soil temperature and decimal latitude in water at Guanica State Forest. There seems to be a negative correlation linear relationship between these with one outlier.

```{r}
# Select the GTDB Taxonomic lineage and separate into taxonomic levels
sankey_data2 <- NEON_MAGs_co %>% 
  select(`GTDB-Tk Taxonomy Lineage`) %>% 
  # NAs are likely Archaea
  replace_na(list(`GTDB-Tk Taxonomy Lineage` = 'Archaea')) %>% 
  # Pavian format requires p__ etc
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";") 

# Fill in the NAs with the taxonomic the higher taxonomic level to the left
sankey_data2[] <- t(apply(sankey_data2, 1, zoo::na.locf))

# Put the data into a format that can be read by the Sankey App

sankey_data2 <- sankey_data2 %>% 
  unite(col = "classification", c(Domain, Phylum, Class, Order, Family, Genus, Species), sep=';') %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") 

# Create format for Pavian with counts for each taxonomic level
sankey_data_s <- sankey_data2
sankey_data_g <- sankey_data2
sankey_data_f <- sankey_data2
sankey_data_o <- sankey_data2
sankey_data_c <- sankey_data2
sankey_data_p <- sankey_data2
sankey_data_d <- sankey_data2

sankey_data_g$classification <- sub("\\|s__.*", "", sankey_data_g$classification)  
sankey_data_f$classification <- sub("\\|g__.*", "", sankey_data_f$classification)  
sankey_data_o$classification <- sub("\\|f__.*", "", sankey_data_o$classification)  
sankey_data_c$classification <- sub("\\|o__.*", "", sankey_data_c$classification)  
sankey_data_p$classification <- sub("\\|c__.*", "", sankey_data_p$classification)  
sankey_data_d$classification <- sub("\\|p__.*", "", sankey_data_d$classification)  

sankey_data_allTaxa <- bind_rows(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) %>% 
  mutate(classification = as.factor(classification)) %>% 
  count(classification) %>% 
# rename for Pavian format
  rename(`#SampleID` = `classification`) %>% 
  rename(`Metaphlan2_Analysis` = `n`) 
```
```

```{r}
knitr::include_url("Exercise 3.html")
```
This Sankey plot shows all the taxa at Guanica State Forest. 

### Taxonmic Group Across Sites 

```{r}
ggtree(tree_Dormibacterota)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tiplab(size=2, hjust=-.1) +
  xlim(0,20) +
  geom_point(mapping=aes(color=`Ecosystem Subtype`)) 
  labs(title = "Dormibacterota vs Ecosystem Subtypes", y = "Ecosystem Subtype", x= "Dormibacterota")
```

This graph showcases the ecosystems in which Dormibacterota reside in. This graph showcases Dormibacterota resides in mostly wetlands and shrublands. 

```{r}
NEON_data_Dormibacterota %>%
  ggplot(aes(x = `Total Number of Bases`, y = `Gene Count`, color = Phylum)) +
  geom_point() +
  coord_flip()
  labs(title = "Total Number of Bases at Phylum", y = "Total Number of Bases", x= "Gene Counts")
```

This graph showcases the total number of bases for Dormibacterota This can be used to determine its prevalance anywhere in the world across all sites. 

```{r}
NEON_data_Dormibacterota %>%   
ggplot(aes(x = fct_infreq(Phylum), y = `Total Number of Bases`)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle=45, vjust=1, hjust=1))
  labs(title = "Total Number of Bases at Various Phylum", y = "Total Number of Bases", x= "Phylum")
```

This graph showcases the total number of bases for Dormibacterota. This can be used to determine the prevelance of this taxonimic group acorss all sites. 



```{r}
NEON_data_Dormibacterota %>%
  ggplot(aes(x = Phylum)) +
  geom_bar() +
  coord_flip()
   labs(title = "Counts at Various Phylum", y = "Phylum", x= "Gene Counts")
```

This graph showcases the gene count of Dormibacterota. It has has an avaerge count compared to other phylums, but is on the lower end of that average. This can be used to determine the prevelance of this taxonimic group acorss all sites. 
 

```{r}
NEON_data_Dormibacterota %>%
  ggplot(aes(x = fct_rev(fct_infreq(Phylum)), fill = Site)) +
  geom_bar() +
  coord_flip()
  
```

Dormibacterota is present at Healy Denali National Park, Natinal Grasslands LBJ, NEON combined assembly, and Wind River Experimental Forest. It is most present National Grasslands LBJ in Texas, USA. 

```{r}
NEON_data_Dormibacterota %>% 
ggplot(aes(x = fct_rev(fct_infreq(Class)), fill = Site)) +
  geom_bar(position = "dodge") +
  coord_flip() +
  labs(title = "MAG Counts for SubPlot", x = "Class", y = "Count") 
```

This graoh is another visualization of the amount of Dormibacterota at various sites. 

```{r}
kable(
NEON_data_Dormibacterota %>% 
  select(c(`GTDB-Tk Taxonomy Lineage`, `Total Number of Bases`)) %>% 
  filter(`Total Number of Bases` > 1000)
)
```

This table is shows the total number of bases for various Dormibacterota. 

```{r}
NEON_FULL <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = c("Sample Name")) %>%
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID"))
```

```{r}
NEON_FULL_D <- NEON_FULL %>%
  filter(str_detect(`Phylum`, "Dormibacterota"))
```


```{r}
NEON_FULL_D %>%   
ggplot(aes(x = `Site.x`, y = `soilTemp`)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle=45, vjust=1, hjust=1))
  labs(title = "Dormibacterota Site.X vs soilTemp", x = "Site.x", y = "Soil Temperature")
```
This graph showcases the soil temps at the various sites where Dormibacterota is found. 

```{r}
ggplot(data = NEON_FULL_D, aes(x = `Ecosystem Subtype`, y = `soilTemp`)) +
    geom_point(aes(color= Order)) +
    labs(title = "Dormibacterota Ecosystem Subtype vs Temperature", x = "Ecosystem Subtype", y = "Soil Temperature")
```

This graph showcases the soil temperature at the various ecosystem subtypes that Dormibacterota is found in. Dormibacterota are shown in green. 

```{r}
NEON_FULL_D %>%
ggplot(aes(x = Genus)) +
  geom_bar(position = position_dodge2(width = 0.9, preserve = "single")) +
  coord_flip()
  labs(title = "Genus vs Count", x = "Count", y = "Genus")
```
This graph showcases the Genus Count in Dormibacterota. 

```{r}
ggplot(data = NEON_FULL_D, aes(x = `Ecosystem Subtype`, y = `soilTemp`)) +
    geom_boxplot()
    labs(title = "Dormibacterota Ecosystem Subtype vs soilTemp", x = "Ecosystem Subtype", y = "soilTemp")
```

This graph shows the soil temperatures of different ecosystem subtypes in Dormibacterota. 

```{r}
NEON_FULL_D %>%   
ggplot(aes(x = fct_infreq(Genus), y = `Geographic Location`)) +
  geom_point() +
  theme(axis.text.x = element_text(angle=45, vjust=1, hjust=1))
  labs(title = "Genus vs Total Number of Bases", x = "Genus", y = "Total Number of Bases")
```
This graph showcases the Genus in various geographic locations for Dormibacterota. 


```{r}
# Select the GTDB Taxonomic lineage and separate into taxonomic levels
sankey_data2 <- NEON_MAGs_co %>% 
  select(`GTDB-Tk Taxonomy Lineage`) %>% 
  # NAs are likely Archaea
  replace_na(list(`GTDB-Tk Taxonomy Lineage` = 'Archaea')) %>% 
  # Pavian format requires p__ etc
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";") 

# Fill in the NAs with the taxonomic the higher taxonomic level to the left
sankey_data2[] <- t(apply(sankey_data2, 1, zoo::na.locf))

# Put the data into a format that can be read by the Sankey App

sankey_data2 <- sankey_data2 %>% 
  unite(col = "classification", c(Domain, Phylum, Class, Order, Family, Genus, Species), sep=';') %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") %>% 
  mutate_at("classification", str_replace, ";", "|") 

# Create format for Pavian with counts for each taxonomic level
sankey_data_s <- sankey_data2
sankey_data_g <- sankey_data2
sankey_data_f <- sankey_data2
sankey_data_o <- sankey_data2
sankey_data_c <- sankey_data2
sankey_data_p <- sankey_data2
sankey_data_d <- sankey_data2

sankey_data_g$classification <- sub("\\|s__.*", "", sankey_data_g$classification)  
sankey_data_f$classification <- sub("\\|g__.*", "", sankey_data_f$classification)  
sankey_data_o$classification <- sub("\\|f__.*", "", sankey_data_o$classification)  
sankey_data_c$classification <- sub("\\|o__.*", "", sankey_data_c$classification)  
sankey_data_p$classification <- sub("\\|c__.*", "", sankey_data_p$classification)  
sankey_data_d$classification <- sub("\\|p__.*", "", sankey_data_d$classification)  

sankey_data_allTaxa <- bind_rows(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) %>% 
  mutate(classification = as.factor(classification)) %>% 
  count(classification) %>% 
# rename for Pavian format
  rename(`#SampleID` = `classification`) %>% 
  rename(`Metaphlan2_Analysis` = `n`) 
```


```{r}
knitr::include_url("Exercise 2.html")
```
This Sankey Plot shows the combined assembilies for Dormibacterota.



## Discussion:

Guanica State Forest was found to have six main phylums with Actinobacteriota being the most prevalent with 30 counts. There have been an additional 30 counts of novel bacteria at this site. The total MAGs for Actinobacteriota found at GUAN were between 20 and 40. There was found to be a negative correlation between the soil temperature and elevation at GUAN where the soil temperature was highest at the lowest elevation and lowest at the highest elevation. This correlates to the type of soil found at Guanica State Forest which are known to be dry and desert-like. The levels of pH in the soil in relation to elevation were inconsistent. Dormibacterota was found in six ecosubtypes mostly in wetlands and shrublands. Dormibacterota has a gene count of about 5000 which is around average compared to other phylums. Dormibacterota is also found mostly at National Grasslands LJB in Texas, USA. Dormibacterota is found in Grasslands where soil temperature is around 15-20C.


## Conclusion:

Guanica State Forest is home to a diverse range of phylums with Actinobacteriota being the most prevelant. Actinobacteriota survives in these dry desert like conditions. Dormibacterota is found in a diverse range of ecosubtypes including wetlands and shrublands, it thrives in more wet soil environments. 


## References:

Ferrari, B. (2021). Persistence and resistance: Survival mechanisms of Candidatus Dormibacterota from nutrient-poor Antarctic soils.

“Guanica Forest Neon / Guan.” Guanica Forest NEON \| NSF NEON \| Open Data to Understand Our Ecosystems,

Montgomery, Kate, Timothy J. Williams, Merryn Brettle, Jonathan F. Berengut, Angelique E. Ray, Eden Zhang, Julian Zaugg, Philip Hugenholtz, and Belinda C. Ferrari. 2021. “Persistence and Resistance: Survival Mechanisms of Candidatus Dormibacterota from Nutrient-Poor 


















